app/src/main/java/woowacourse/shopping/ui/ViewModelFactory.kt

```diff
@@ -0,0 +1,22 @@
+package woowacourse.shopping.ui
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import woowacourse.shopping.data.di.Container
+import kotlin.reflect.KClass
+import kotlin.reflect.full.primaryConstructor
+
+class ViewModelFactory : ViewModelProvider.Factory {
+    override fun <T : ViewModel> create(modelClass: Class<T>): T {
+        val constructor =
+            modelClass.kotlin.primaryConstructor
+                ?: throw IllegalArgumentException("Unknown ViewModel")
+
+        val params =
+            constructor.parameters.map { parameter ->
+                Container.instance<T>(parameter.type.classifier as KClass<*>)
+            }.toTypedArray()
```

Comment:
```suggestion
        val params =
            constructor.parameters.map { parameter ->
                val parameterType = parameter.type.classifier as KClass<*>
                Container.instance<T>(parameterType)
            }.toTypedArray()
```

`parameter.type.classifier as KClass<*>` 를 변수로 선언하면 가독성이 더 좋아질 것 같습니다! 
위와 같이 변경하면 ` KClass<*>`로 캐스팅되지 않았을 때의 처리도 더 간편하게 할 수 있을 것 같아요!

Reply:
> parameter.type.classifier as KClass<*>에서 KClass<*>로 캐스팅되지 않는 상황

해나가 위에서 리뷰 남겨주신 내용을 분기처리 하여 해결하면서 변수로 같이 선언했더니 가독성이 더 좋아졌네요 !! 👍